classes/class.ilObjShortLink.php

```diff
@@ -0,0 +1,303 @@
+<?php
+/*
+	+-----------------------------------------------------------------------------+
+	| ILIAS open source                                                           |
+	+-----------------------------------------------------------------------------+
+	| Copyright (c) 1998-2014 ILIAS open source, University of Cologne            |
+	|                                                                             |
+	| This program is free software; you can redistribute it and/or               |
+	| modify it under the terms of the GNU General Public License                 |
+	| as published by the Free Software Foundation; either version 2              |
+	| of the License, or (at your option) any later version.                      |
+	|                                                                             |
+	| This program is distributed in the hope that it will be useful,             |
+	| but WITHOUT ANY WARRANTY; without even the implied warranty of              |
+	| MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               |
+	| GNU General Public License for more details.                                |
+	|                                                                             |
+	| You should have received a copy of the GNU General Public License           |
+	| along with this program; if not, write to the Free Software                 |
+	| Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA. |
+	+-----------------------------------------------------------------------------+
+*/
+
+require_once('./Customizing/global/plugins/Services/UIComponent/UserInterfaceHook/ShortLink/classes/class.ilShortLinkAccess.php');
+
+
+/**
+ * Class ilObjShortLink
+ * Object for the ShortLinkPlugin
+ *
+ * @author  Tomasz Kolonko <thomas.kolonko@ilub.unibe.ch>
+ * @version $Id$
+ */
+class ilObjShortLink {
+
+    /**
+     * @var ilDB $db
+     */
+    protected $db;
+    /**
+     * @var ilObjUser $usr
+     */
+    protected $usr;
+    /**
+     * @var ilShortLinkPlugin $pl
+     */
+    protected $pl;
+    /**
+     * @var int $id
+     */
+    protected $id = 0;
+    /**
+     * @var string $contact
+     */
+    protected $contact;
+    /**
+     * @var string $longURL
+     */
+    protected $longURL;
+    /**
+     * @var string $shortLink
+     */
+    protected $shortLink;
+    /**
+     * @var $customer
+     */
+    protected $customer;
+    /**
+     * @var $currentUserId;
+     */
+    protected $currentUserId;
+
+
+    public function __construct() {
```

Comment:
/**
     * ilObjShortLink constructor
     *
     * Makes global DB and User variable locally available trough member variables.
     * Creates new ilShortLinkPlugin and makes it locally available. 
     */